### ISSUE TITLE
Enhancement: Reset song/week scores

### ISSUE THREAD

**Issue by LukerUpgradez** [FunkinCrew/Funkin#2708]
Now that I'm working towards improving my scores I've noticed that this feature is missing. Currently the only ways to change your scores that I'm aware of are directly manipulating the json file and deleting your save data, neither of which are ideal. Being able to set your score for a week in story mode or a song in freeplay back to 0 in-game would be a great convenience.

The main reason for a quick reset would be to keep up with new versions of charts. Say that you release a chart that has a triple, and a player gets a gold P rank on it. But then the charter decides that triples are lame, so in the next version they make it a double. Now the song has one less note, and the maximum possible score is one note smaller. That means the player who got a gold P rank on the first version will likely never be able to beat their saved score on the second. A reset button would let that player go for a legit perfect score on the new chart without any hassle.

It's also nice future-proofing for modded weeks that can have volatile developments. For example, I played a mod whose release version had a week with five songs in it. In an update, they moved two of those songs to their own week and left the original week with three. That left the score I got on the old version several hundred thousand points higher than the new maximum. This isn't an issue I expect the official game to deal with, but the feature would be a nice safeguard against big changes like that.

There's also just some value in getting a perfect score and then saying "Screw it, I'm gonna do it all over again just to prove that I can".

I don't know if this is already in the cards (I couldn't find any open matching issue) but I thought I'd put it forward.

**Comment by Hundrec** [comment]
This suggestion is excellent and deserves some consideration after over four months have passed.

Here are just a few charting issues that would affect song scores across different versions:
* Spookeez https://github.com/FunkinCrew/Funkin/issues/2652
* High Erect https://github.com/FunkinCrew/Funkin/issues/2689
* Cocoa Erect (fixed) https://github.com/FunkinCrew/Funkin/issues/3332
* Philly Nice https://github.com/FunkinCrew/Funkin/issues/3386

Additionally, full-on recharts could always happen, increasing the need for a score reset feature. Perhaps this could initially be implemented as a debug hotkey with a confirmation safeguard.

**Comment by LukerUpgradez** [comment]
A hotkey that opens a confirmation/'Are you sure you want to delete this score?" message was about how I expected an implementation would work.

When I first wrote this suggestion, I was unsure how you'd design the pop-up for that message, but now that we're in 5.0, the instrumental selection pop-up is a near-perfect template. The team could probably reuse most of the code from it.

**Comment by Lasercar** [comment]
Why are all the interesting enhancements several pages in? Just visualising this enhancement right now makes me want to try implementing it.

**Comment by Average-FNF-Modder** [comment]
This is so needed. I think I tried to ask for this once though, and it was denied. Keep pushing.

**Comment by MistrDJ** [comment]
I hope this option is considered for the future, it would be a big convenience, especially as someone who loves to grind out songs for new hiscores
i noticed to get the hey and cheer animations workin in bopeebo and spookeez pico in Pitstop 2 notes were removed from the chart that were previously there in the absence of those animations
but that means that its impossible for me to beat my hiscore on those songs because i got those scores making use of notes that dont exist anymore, my hiscores are unbeatable

**Comment by Average-FNF-Modder** [comment]
> I hope this option is considered for the future, it would be a big convenience, especially as someone who loves to grind out songs for new hiscores i noticed to get the hey and cheer animations workin in bopeebo and spookeez pico in Pitstop 2 notes were removed from the chart that were previously there in the absence of those animations but that means that its impossible for me to beat my hiscore on those songs because i got those scores making use of notes that dont exist anymore, my hiscores are unbeatable

Honestly I don't really like them removing those notes anyways, instead of making it impossible to have an animation play during a note, they should just make a dedicated 'Play Animation' note class or something like that. Maybe I'll make an issue about it. This would still be useful for any other chart changes ofc. Maybe they could even have your score automatically reset if a songs chart was updated in a new patch.

**Comment by Hundrec** [comment]
Yeah, I'd also prefer if the player had inputs for special animations. Makes it feel like we triggered them

**Comment by Average-FNF-Modder** [comment]
> > I hope this option is considered for the future, it would be a big convenience, especially as someone who loves to grind out songs for new hiscores i noticed to get the hey and cheer animations workin in bopeebo and spookeez pico in Pitstop 2 notes were removed from the chart that were previously there in the absence of those animations but that means that its impossible for me to beat my hiscore on those songs because i got those scores making use of notes that dont exist anymore, my hiscores are unbeatable
> 
> Honestly I don't really like them removing those notes anyways, instead of making it impossible to have an animation play during a note, they should just make a dedicated 'Play Animation' note class or something like that. Maybe I'll make an issue about it. This would still be useful for any other chart changes ofc. Maybe they could even have your score automatically reset if a songs chart was updated in a new patch.

Side note, this would require having note types allow for an options panel like events, which is also something I think would be useful, I'm definitely going to write an issue about this now.

**Comment by LukerUpgradez** [comment]
Huh, I just noticed that this feature was added to the [rejected features list](docs/CONTRIBUTING.md). It references #3916 instead of this issue, which has a different explanation for why the feature should be added.

I don't think that's a reason to close this issue, as the linked justification for not adding this feature is strictly from a creator's perspective and not a regular player's. As Mistr pointed out, version inconsistencies are now an issue that affects base game songs, so this is something that impacts all players, not just mod-makers.

If the team is determined not to implement an explicit reset button, maybe an alternative could be that scoring could recognize version control, and your score auto-resets if a chart is updated? I don't think that's a good idea myself, since it means casual players can have scores they earned arbitrarily wiped based on tiny updates they may not even notice, but it's a thought.

I agree with the above conversation regarding better tying animations and notes together, but that probably should be its own issue, if it isn't one already.

**Comment by Michu0204** [comment]
Please reconsider this we know we ALL need this 😭
